test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -8464,42 +8464,71 @@ def cleanup_jobs(self):
         jobs. Specifying an extend parameter could override
         this behavior.
         """
+        delete_xt = 'nomailforce'
         select_xt = None
         if self.is_history_enabled():
+            delete_xt += 'deletehist'
             select_xt = 'x'
         jobs = self.status(JOB, extend=select_xt)
-        job_ids = [x['id'] for x in jobs]
-
-        if len(job_ids) < 1:
-            return
+        job_ids = sorted(list(set([x['id'] for x in jobs])))
+        host_pid_map = {}
+        for job in jobs:
+            exec_host = job.get('exec_host', None)
+            if not exec_host or 'session_id' not in job:
+                continue
+            _host = exec_host.split('/')[0].split(':')[0]
+            if _host not in host_pid_map:
+                host_pid_map.setdefault(_host, [])
+            host_pid_map[_host].append(job['session_id'])
 
         # Turn off scheduling so jobs don't start when trying to
         # delete. Restore the orignial scheduling state
         # once jobs are deleted.
-        sched_to_start = []
+        sched_state = []
         scheds = self.status(SCHED)
         for sc in scheds:
             if sc['scheduling'] == 'True':
-                sched_to_start.append(sc['id'])
-            # runas is required here because some tests remove
-            # current user from managers list
-            a = {'scheduling': 'False'}
-            self.manager(MGR_CMD_SET, SCHED, a, id=sc['id'],
-                         runas=ROOT_USER)
-
-        qdel_path = os.path.join(self.pbs_conf["PBS_EXEC"], "bin", "qdel")
-        if select_xt == 'x':
-            cmd = [qdel_path, "-x"] + job_ids
-        else:
-            cmd = [qdel_path] + job_ids
-
-        self.du.run_cmd(cmd=cmd)
-
+                sched_state.append(sc['id'])
+                # runas is required here because some tests remove
+                # current user from managers list
+                a = {'scheduling': 'False'}
+                self.manager(MGR_CMD_SET, SCHED, a, id=sc['id'],
+                             runas=ROOT_USER)
+                self.expect(SCHED, a, id=sc['id'])
```

Comment:
I'm not sure what this accomplishes.  You just set it to false, won't manager throw an exception if it fails?